### PR TITLE
Missing tests for IdSet

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by inkydragon — APPROVED** [review]
before pr: https://app.codecov.io/gh/JuliaLang/julia/blob/master/base%2Fidset.jl

File: test/sets.jl [hunk 1]
```diff
@@ -1037,6 +1037,8 @@ end
     @test !isempty(A)
     A = empty!(A)
     @test isempty(A)
+    @test isnothing(sizehint!(A, 10))
```
Comment (DilumAluthge) [inline test/sets.jl:1040]: Is there a better way to test this `sizehint!()` call? Maybe by looking at the length of `.ht` before and after the `sizehint!()` call?

Comment (inkydragon) [reply test/sets.jl:1040]: Maybe add a new `TODO`?

https://github.com/JuliaLang/julia/blob/10f294b422ce56759639e99399aec5c407bafc00/base/idset.jl#L81-L93

Comment (DilumAluthge) [reply test/sets.jl:1040]: Ah, I didn't realize it was currently a no-op.

Comment (DilumAluthge) [inline test/sets.jl:1040]:
```suggestion
    @test isnothing(sizehint!(A, 10)) # Currently a no-op. TODO: Add a better test once the functionality has been implemented.
```

**Review by DilumAluthge — APPROVED** [review]

**Pull request merged**